torch/_prims/__init__.py

```diff
@@ -1903,10 +1903,10 @@ def _where_meta(
 ) -> TensorLikeType:
 
     return _elementwise_meta(
+        pred.to(a.dtype),
```

Comment:
while this nominally fixes the strides problem, this is conceptually a very confusing thing to do, and it would wrong to wrong results if `a` was a 0d tensor, and `pred` was a 1+D tensor this would lead to wrong type promotion (now `a's` dtype would have a higher precendence than it should).

Reply:
Ok, given that `where` prim is the only one using `args_with_fixed_dtypes` I suggest modifying `_elementwise_meta` to prepend rather than append `args_with_fixed_dtypes`, that should have correct results too.

Reply:
fixed, as suggested.